pkg/services/annotations/accesscontrol/accesscontrol_test.go

```diff
@@ -0,0 +1,129 @@
+package accesscontrol
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/grafana/grafana/pkg/components/simplejson"
+	"github.com/grafana/grafana/pkg/infra/db"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	"github.com/grafana/grafana/pkg/services/annotations"
+	"github.com/grafana/grafana/pkg/services/annotations/testutil"
+	"github.com/grafana/grafana/pkg/services/dashboards"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"github.com/grafana/grafana/pkg/services/user"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	dashScopeType = annotations.Dashboard.String()
+	orgScopeType  = annotations.Organization.String()
+)
+
+func TestIntegrationAuthorize(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	sql := db.InitTestDB(t)
+
+	authz := NewAuthService(sql, featuremgmt.WithFeatures())
+
+	dash1 := testutil.CreateDashboard(t, sql, featuremgmt.WithFeatures(), dashboards.SaveDashboardCommand{
+		UserID: 1,
+		OrgID:  1,
+		Dashboard: simplejson.NewFromAny(map[string]any{
+			"title": "Dashboard 1",
+		}),
+	})
+
+	dash2 := testutil.CreateDashboard(t, sql, featuremgmt.WithFeatures(), dashboards.SaveDashboardCommand{
+		UserID: 1,
+		OrgID:  1,
+		Dashboard: simplejson.NewFromAny(map[string]any{
+			"title": "Dashboard 2",
+		}),
+	})
+
+	u := &user.SignedInUser{
+		UserID: 1,
+		OrgID:  1,
+	}
+	role := testutil.SetupRBACRole(t, sql, u)
+
+	type testCase struct {
+		name              string
+		permissions       map[string][]string
+		expectedResources *AccessResources
+		expectedErr       error
+	}
+
+	testCases := []testCase{
+		{
+			name: "should have both scopes and all dashboards",
+			permissions: map[string][]string{
+				accesscontrol.ActionAnnotationsRead: {accesscontrol.ScopeAnnotationsAll},
+				dashboards.ActionDashboardsRead:     {dashboards.ScopeDashboardsAll},
+			},
+			expectedResources: &AccessResources{
+				Dashboards: map[string]int64{dash1.UID: dash1.ID, dash2.UID: dash2.ID},
+				ScopeTypes: map[any]struct{}{dashScopeType: {}, orgScopeType: {}},
+			},
+		},
+		{
+			name: "should have only organization scope and no dashboards",
+			permissions: map[string][]string{
+				accesscontrol.ActionAnnotationsRead: {accesscontrol.ScopeAnnotationsTypeOrganization},
```

Comment:
nit: it might be good to also add some read permissions on dashboard for this test and check that dashboard list doesn't get populated if the user is missing the permissions to read dashboard annotations.